cosmos/x/evm/plugins/txpool/mempool/mempool.go

```diff
@@ -22,74 +22,55 @@ package mempool
+	*txpool.TxPool
 
-	// NonceRetriever is used to retrieve the nonce for a given address (this is typically a
-	// reference to the StateDB).
-	nr NonceRetriever
+	// serializer converts eth txs to sdk txs when being iterated over.
+	serializer SdkTxSerializer
 
-	// ethTxCache caches transactions that are added to the mempool so that they can be retrieved
-	// later
-	ethTxCache map[common.Hash]*coretypes.Transaction
+	// cp is used to retrieve the current chain config.
+	cp ConfigurationPlugin
 
-	// nonceToHash maps a nonce to the hash of the transaction that was added to the mempool with
-	// that nonce. This is used to retrieve the hash of a transaction that was added to the mempool
-	// by nonce.
-	nonceToHash map[common.Address]map[uint64]common.Hash
+	// block data for the current block.
+	blockNumber *big.Int
+	blockTime   uint64
 
-	// We have a mutex to protect the ethTxCache and nonces maps since they are accessed
-	// concurrently by multiple goroutines.
-	mu sync.RWMutex
+	// baseFee is the base fee for priority calculation.
+	baseFee *big.Int
 }
 
-// NewPolarisEthereumTxPool creates a new Ethereum transaction pool.
-func NewPolarisEthereumTxPool() *EthTxPool {
-	tpp := EthereumTxPriorityPolicy{
-		baseFee: big.NewInt(0),
-	}
-	config := mempool.PriorityNonceMempoolConfig[*big.Int]{
-		TxReplacement: EthereumTxReplacePolicy[*big.Int]{
-			PriceBump: 10, //nolint:gomnd // 10% to match geth.
-		}.Func,
-		TxPriority: mempool.TxPriority[*big.Int]{
-			GetTxPriority: tpp.GetTxPriority,
-			Compare: func(a *big.Int, b *big.Int) int {
-				return a.Cmp(b)
-			},
-			MinValue: big.NewInt(-1),
-		},
-		MaxTx: 10000, //nolint:gomnd // todo: parametize this.
-	}
+// NewWrappedGethTxPool creates a new Ethereum transaction pool.
+func NewWrappedGethTxPool() *WrappedGethTxPool {
+	return &WrappedGethTxPool{}
+}
 
-	return &EthTxPool{
-		PriorityNonceMempool: mempool.NewPriorityMempool(config),
-		nonceToHash:          make(map[common.Address]map[uint64]common.Hash),
-		ethTxCache:           make(map[common.Hash]*coretypes.Transaction),
-		priorityPolicy:       &tpp,
-	}
+// SetTxPool sets the underlying Geth TxPool.
+func (gtp *WrappedGethTxPool) SetTxPool(txPool *txpool.TxPool) {
+	gtp.TxPool = txPool
 }
 
-// SetNonceRetriever sets the nonce retriever db for the mempool.
-func (etp *EthTxPool) SetNonceRetriever(nr NonceRetriever) {
-	etp.nr = nr
+// Setup sets the chain config and sdk tx serializer on the wrapped Geth TxPool.
+func (gtp *WrappedGethTxPool) Setup(cp ConfigurationPlugin, serializer SdkTxSerializer) {
+	gtp.cp = cp
+	gtp.serializer = serializer
 }
 
-// SetBaseFee updates the base fee in the priority policy.
-func (etp *EthTxPool) SetBaseFee(baseFee *big.Int) {
-	etp.priorityPolicy.baseFee = baseFee
+// Prepare updates the mempool for the current block. Set the block number, block time, and base
+// fee.
+func (gtp *WrappedGethTxPool) Prepare(blockNumber *big.Int, blockTime uint64, baseFee *big.Int) {
```

Comment:
why not just hook this up to the NewChainHead channel?

Reply:
that gets emitted after the end of the prev block, in which case the current block data has already updated.